pandas/tests/arrays/test_datetimes.py

```diff
@@ -15,6 +15,18 @@
 
 
 class TestDatetimeArrayConstructor:
+
+    def test_only_1dim_accepted(self):
+        arr = np.array([0, 1, 2, 3], dtype='M8[h]').astype('M8[ns]')
+
+        with pytest.raises(ValueError, match="Only 1-dimensional"):
+            # 2-dim
+            DatetimeArray(arr.reshape(2, 2))
```

Comment:
To be clear: this already fails currently, right? You are mainly catching the error early / providing a better error message ?

Reply:
No, this is currently accepted incorrectly.

Reply:
With latest master:

```
In [15]: pd.__version__
Out[15]: '0.25.0.dev0+791.gf0919f272'

In [16]: arr = np.array([0, 1, 2, 3], dtype='M8[h]').astype('M8[ns]') 

In [17]: pd.arrays.DatetimeArray(arr.reshape(2, 2)) 
...
~/scipy/pandas/pandas/_libs/tslibs/conversion.pyx in pandas._libs.tslibs.conversion.convert_to_tsobject()

TypeError: Cannot convert input [['1970-01-01T00:00:00.000000000' '1970-01-01T01:00:00.000000000']] of type <class 'numpy.ndarray'> to Timestamp
```

Reply:
Try changing [17] to `res = pd.arrays.DatetimeArray(arr.reshape(2, 2))`.  I'm pretty sure that error is coming from an attempt to call `__repr__`.

Reply:
ah .. yes :-)